__tests__/components/SignerSelector.spec.ts

```diff
@@ -0,0 +1,121 @@
+        describe('SignerBaseFilter', () => {
+            test('renders signer base filter component when parent signer exists', () => {
+                // Arrange:
+                const wrapper = getSignerSelectorWrapper({
+                    rootSigner: {
+                        label: '',
+                        address: simpleWallet1.address,
+                        multisig: false,
+                        requiredCosigApproval: 0,
+                        parentSigners: [
+                            {
+                                label: 'label',
+                                address: simpleWallet2.address,
+                                multisig: false,
+                                requiredCosigApproval: 0,
+                            },
+                        ],
+                    },
+                });
+
+                // Act + Assert:
+                expect(wrapper.findComponent(SignerBaseFilter).exists()).toBe(true);
+            });
+
+            test('hide signer base filter component when parent signer does not exist', () => {
+                // Arrange:
+                const wrapper = getSignerSelectorWrapper();
+
+                // Act + Assert:
+                expect(wrapper.findComponent(SignerBaseFilter).exists()).toBe(false);
+            });
+
+            test('display label when root signer is multisig and parent signer does not exist', () => {
+                // Arrange:
+                const wrapper = getSignerSelectorWrapper({
+                    rootSigner: {
+                        label: 'label',
+                        address: simpleWallet1.address,
+                        multisig: true,
+                        requiredCosigApproval: 0,
+                        requiredCosigRemoval: 0,
+                    },
+                });
+
+                // Act + Assert:
+                expect(wrapper.find('.signer-selector-single-signer-container').text()).toContain('label');
+                expect(wrapper.find('.signer-selector-single-signer-container').text()).toContain('(Multisig)');
+            });
+        });
+    });
+
+    describe('chosenSigner', () => {
+        test('returns current signer', () => {
+            // Arrange:
+            const wrapper = getSignerSelectorWrapper({
+                value: simpleWallet1.address.plain(),
+            });
+
+            const vm = wrapper.vm as SignerSelectorTs;
+
+            // Act + Assert:
+            expect(vm.chosenSigner).toEqual(simpleWallet1.address.plain());
+        });
+
+        test('set new chosen signer when new signer provided', () => {
+            // Arrange:
+            const newSigner = simpleWallet2.address.plain();
+
+            const wrapper = getSignerSelectorWrapper();
+
+            const vm = wrapper.vm as SignerSelectorTs;
+
+            // Act:
+            vm.onSignerSelectorChange(newSigner);
+
+            // Assert:
+            expect(wrapper.emitted('input')).toEqual([[newSigner]]);
```

Comment:
should you check vm.chosenSigner too ?

Reply:
this one is interesting,  the `vm.chosenSigner` is not updated in the unit test, but it works in production.
I will dig more on this.

Reply:
After doing some digging, and discussing with the team. We are safe to **skip** `check vm.chosenSigner` in this case.

Vue `$emit` event is used to passing date from `child` to `parent`. So in this case `wrapper.emitted('input')` is enough to verify.